### PR TITLE
Adding Azure Monitor Pipelines public preview.

### PR DESCRIPTION
**Pull request by andborja**
Work item:
https://msazure.visualstudio.com/One/_workitems/edit/27558925

# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

> [!NOTE]
> As of January 2024 there is no PR assignee. This is expected. See https://aka.ms/azsdk/pr-arm-review.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary>Click here to see the details of Step 1, Breaking Changes review</summary>

If you are in purview of Step 1 of the diagram, follow the [Breaking Changes review process].  
**IMPORTANT!** This applies even if you believe your PR was mislabeled, for any reason, including tool failure.
</details>

<details> 
<summary>Click here to see the details of Step 2, ARM review</summary>

See https://aka.ms/azsdk/pr-arm-review.
</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] See [ARM review queue] (for **PR merge** queues, see [2]).  
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])  
The ARM reviewer on-call engineer visits the merge queue twice a day, so the approximate ETA for merges is 12 - 24 hours.

</details>

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [ ] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>

This is a copy of the private validated version in RPSaaSMaster: 

https://github.com/Azure/azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/monitor/resource-manager/Microsoft.Monitor/preview/2023-10-01-preview

This is in preparation for public preview of Azure Monitor Pipelines.

<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with PR workflow diagram Step 2 (ARM review), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://aka.ms/azsdk/pr-arm-review#about-the-arm-review-queue
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[Breaking Changes review process]: https://aka.ms/brch

### CODE REVIEW

**Comment by andborja** [comment]
Thanks @raosuhas for your attention.
This is a copy of the version in private branch RPSaaSMaster at https://github.com/Azure/azure-rest-api-specs-pr/commit/0e581ddf8cc0717779292981f882ed4b93f25143. The changes on this PR fixes typespec new validations, nothing for the business.
The same fixes are under review for RPSaaSMaster as part of PR https://github.com/Azure/azure-rest-api-specs-pr/pull/17852.

The version we are moving to this main public branch rewrites in typespec the current version in main https://github.com/Azure/azure-rest-api-specs/tree/main/specification/monitor/resource-manager/Microsoft.Monitor/stable/2023-04-03, and adds the new resource type pipelineGroups.

**Comment by andborja** [comment]
@raosuhas, I cannot edit the labels for this PR, so please remove the ARMChangesRequested label when possible or take another look at this pr when you have some time.
Thanks!

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2023-10-01-preview/azuremonitor.json [hunk 1]
```diff
@@ -0,0 +1,1980 @@
...
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Monitor/pipelineGroups": {
+      "get": {
+        "operationId": "PipelineGroups_ListBySubscription",
+        "tags": [
+          "PipelineGroups"
+        ],
+        "description": "Lists all workspaces in the specified subscription",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Monitor/preview/2023-10-01-preview/azuremonitor.json:137]: >v3 [](http://example.com/codeflow?start=69&length=2)

v5 . here and everywhere else.

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2023-10-01-preview/azuremonitor.json [hunk 2]
```diff
@@ -0,0 +1,1980 @@
...
+        }
+      },
+      "required": [
+        "receivers",
+        "processors",
+        "exporters",
+        "service"
+      ]
+    },
+    "PipelineGroupUpdate": {
```
Comment (ramoka178) [inline specification/monitor/resource-manager/Microsoft.Monitor/preview/2023-10-01-preview/azuremonitor.json:1357]: >PipelineGroupUpdate [](http://example.com/codeflow?start=5&length=19)

can the same object PipelineGroup as PUT be used ?

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2023-10-01-preview/azuremonitor.json (hunk 1 above)
Comment (andborja) [reply specification/monitor/resource-manager/Microsoft.Monitor/preview/2023-10-01-preview/azuremonitor.json:137]: This is how it is defined for that resource type (another team) in the original base (https://github.com/Azure/azure-rest-api-specs/blob/main/specification/monitor/resource-manager/Microsoft.Monitor/stable/2023-04-03/monitoringAccounts_API.json).
This PR does not make any changes to that resource type 'accounts', only re-write it in typespec.

File: specification/monitor/resource-manager/Microsoft.Monitor/preview/2023-10-01-preview/azuremonitor.json (hunk 2 above)
Comment (andborja) [reply specification/monitor/resource-manager/Microsoft.Monitor/preview/2023-10-01-preview/azuremonitor.json:1357]: That is generated by typespec. I think it is in purpose to support future non-updatable attributes.

**Comment by andborja** [comment]
Thanks @ramoka178 for your review. I don't have permissions to remove the label. I hope you can review again this PR after my answers.

**Comment by andborja** [comment]
Please keep in mind that this version was reviewed and approved before for private branch RPSaaSMaster:
https://github.com/Azure/azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/monitor/Microsoft.Monitor
This is just moving it to main public branch.

**Comment by rkmanda** [comment]
pl fix these errors - https://github.com/Azure/azure-rest-api-specs/pull/28620/checks?check_run_id=23582437194

**Comment by rkmanda** [comment]
There are conflicts on your branch, pl resolve them

**Comment by andborja** [comment]
> pl fix these errors - https://github.com/Azure/azure-rest-api-specs/pull/28620/checks?check_run_id=23582437194

Thanks @rkmanda for your review.

Those errors are not mandatory. This is a preview version, so I don't want to make it default. What is your recommendation for it?

**Comment by andborja** [comment]
Hi @rkmanda, it looks like the fix to conflicts requires a new review =(.

Please take a look when you have some time.

Thanks

**Comment by raych1** [comment]
Approved Go sdk breaking change as they are generated on `package-preview-2023-09` tag and has nothing to do with this PR changes.

**Review by rkmanda — APPROVED** [review]

**Pull request merged**